web/book/src/internals/modules.md

```diff
@@ -0,0 +1,95 @@
+# Modules
+
+PRQL needs modules that would:
+
+- allow importing declarations from other files,
+- have namespaces for things like `std`,
+- have hierarchical structure so we can represent files in directories,
+- have unambiguous module structure within a project,
+- be able to compile individual files that are a part of a project.
+
+## Basic definition
+
+Module is a namespace that contains declarations.
+
+Modules can be defined with mod keyword and a code block encased in curly
+braces:
+
+```
+mod my_module {
+    let bangers = (from tracks | take 10)
+}
```

Comment:
Syntax is similar to current discussion on tuples/arrays, but I don't think it will clash on the parser-level. That's because this is on the "level" of the statements and those two are both on the "level" of expressions.

Reply:
I was going to ask about that. Are modules not also structs where the keys are names and the values are either (sub)modules, relations or functions?

Shouldn't we therefore be consistent in the use of symbols and use `[]` possibly rather than `{}`? Your current proposal reads a lot more naturally ofc but curious about the consistency angle.

Reply:
No, semantically modules are not equivalent to structs. They don't have a type, cannot be referred to as a value and have different parsing rules.

So I think that here it's ok to choose the syntax that looks best, instead of seeking consistency with structs.

Reply:
(also moot if we use just files / https://github.com/PRQL/prql/pull/2129#issuecomment-1468861960)